sonar-text-plugin/src/test/java/org/sonar/plugins/secrets/api/SecretMatcherAssert.java

```diff
@@ -0,0 +1,64 @@
+/*
+ * SonarQube Text Plugin
+ * Copyright (C) 2021-2023 SonarSource SA
+ * mailto:info AT sonarsource DOT com
+ *
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License as published by the Free Software Foundation; either
+ * version 3 of the License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this program; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ */
+package org.sonar.plugins.secrets.api;
+
+import java.util.Objects;
+import java.util.Set;
+import java.util.function.BiPredicate;
+import java.util.regex.Pattern;
+import org.assertj.core.api.AbstractAssert;
+
+public class SecretMatcherAssert extends AbstractAssert<SecretMatcherAssert, SecretMatcher> {
+
+  static Set<String> strings = Set.of(
+    "candidate secret with low entropy",
+    "candidate secret with low entropy and patternNot:EXAMPLEKEY",
+    "candidate secret with high entropy: lasdij2338f,.q29cm2acasd has patternNot:EXAMPLEKEY",
+    "candidate secret with high entropy: lasdij2338f,.q29cm2acasd"
+  );
```

Comment:
I don't get the hardcoded quite specific strings here.

Reply:
@nils-werner-sonarsource As there is no equals for predicates I wanted to test that they behave the same. As I am testing postFilter I put strings in there,which I knew would evaluate to true / false based on the actual filter